util/status.cc

```diff
@@ -25,7 +25,7 @@ const char* Status::CopyState(const char* state) {
   ret = strncpy_s(result, cch, state, cch - 1);
```

Comment:
Shouldn't the change be made here as well?

Reply:
According to the strncpy_s documentation, it automatically appends a NULL terminator if there is room in the destination, which is the case here.

Reply:
Wow, that's such a tiny detail buried deep in the documentation! But you are right, this seems like the right way to use strncpy_s. 

>Unlike strncpy, strncpy_s does not pad the destination array with zeroes, This is a common source of errors when converting existing code to the bounds-checked version.
Although truncation to fit the destination buffer is a security risk and therefore a runtime constraints violation for strncpy_s, it is possible to get the truncating behavior by specifying count equal to the size of the destination array minus one: it will copy the first count bytes and append the null terminator as always: `strncpy_s(dst, sizeof dst, src, (sizeof dst)-1);`

What made me initially comment on this is that we now have two way of achieving the same thing but with visible differences (like the copy-count passed in to strncpy*()). It wasn't immediately obvious that they are doing the same thing.

Reply:
Yup. I actually didn't consider it until I saw your comment. It caught me by surprise as well. There is no real bug in non-Windows code either, but I guess gcc-8.1.1 is smart enough to realize there may be a potential bug in the call to strncpy(), but not smart enough to realize we explicitly set the NULL terminator earlier!